exercises/src/main/scala/fpinscala/gettingstarted/GettingStarted.scala

```diff
@@ -153,13 +165,13 @@ object PolymorphicFunctions {
   // Note that `=>` associates to the right, so we could
   // write the return type as `A => B => C`
   def curry[A,B,C](f: (A, B) => C): A => (B => C) =
-    ???
+    (a: A) => (b: B) => f(a, b)
 
   // NB: The `Function2` trait has a `curried` method already
 
   // Exercise 4: Implement `uncurry`
   def uncurry[A,B,C](f: A => B => C): (A, B) => C =
-    ???
+    (a: A, b: B) => f(a)(b)
 
   /*
   NB: There is a method on the `Function` object in the standard library,
```

Comment:
Judging by the note, I think they were expecting something simpler like this:

```
scala> def curry[A,B,C](f: (A, B) => C): A => (B => C) = f.curried
curry: [A, B, C](f: (A, B) => C)A => (B => C)

scala> def uncurry[A,B,C](f: A => B => C): (A, B) => C = Function.uncurried(f)
uncurry: [A, B, C](f: A => (B => C))(A, B) => C

scala> def compose[A,B,C](f: B => C, g: A => B): A => C = f compose g
compose: [A, B, C](f: B => C, g: A => B)A => C
```

The end result would be the same though.